src/ui/components/componentmanager.js

```diff
@@ -177,4 +186,24 @@ export default class ComponentManager {
   getActiveComponent (type) {
     return this._activeComponents.find(c => c.constructor.type === type);
   }
+
+  /**
+   * Returns a list of all names associated with a given component type
+   * @param {string} type The type of the component
+   * @returns {string[]} The component names for the component type
+   */
+  getComponentNamesForComponentType (type) {
+    return this._componentTypeToComponentNames[type];
+  }
+
+  /**
+   * Returns a concatenated list of all names associated with the given component types
+   * @param {string[]} type The types of the component
+   * @returns {string[]} The component names for the component types
+   */
+  getComponentNamesForComponentTypes (types) {
```

Comment:
If someone calls this with a single type, it's equivalent to a call to getComponentNamesForComponentType, right? Because of that, I'd say lets just keep this method and get rid of the other one. It doesn't feel like the other method is adding much.

Reply:
Done